noir-projects/noir-contracts/contracts/token_contract/src/types/token_note.nr

```diff
@@ -40,7 +42,7 @@ impl NoteInterface<TOKEN_NOTE_LEN> for TokenNote {
 
     fn compute_nullifier_without_context(self) -> Field {
         let note_hash_for_nullify = compute_note_hash_for_consumption(self);
-        let secret = get_app_nullifier_secret_key(self.owner);
+        let secret = get_nsk_app_with_npk_m_h(self.owner_npk_m_h);
```

Comment:
The diff between the with and without context seems like something that can easily be turbo pain for people, e.g., you use the wrong, time to get rekt. As I recall, the reasoning was that we need the dev to specify getting the secret, but it seems like such a footgun when we have these two cases. Would be great if we could get rid of the `without_context` case, this was mainly needed for the `compute_note_hash_and_nullifier` as I recall, but there must be something better 🤔

Reply:
Good catch Lasse - This is like `get_notes()` and `view_notes()` saga all over again

Reply:
Naming generally need to be better handling for how we do oracles. Wen `unsafe` in noir 😆